packages/backend/src/links/links.service.ts

```diff
@@ -1,45 +1,71 @@
-import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
-import { CreateLinkDto } from './dto/create-link.dto';
-import { UpdateLinkDto } from './dto/update-link.dto';
-import { PrismaService } from 'nestjs-prisma';
-import { Prisma } from '@prisma/client';
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common'
+import { Prisma } from '@prisma/client'
+import { PrismaService } from 'nestjs-prisma'
+import { CreateLinkDto } from './dto/create-link.dto'
+import { UpdateLinkDto } from './dto/update-link.dto'
+import axios from 'axios'
 
 @Injectable()
 export class LinksService {
   constructor(private readonly prisma: PrismaService) {}
+
   async create(createLinkDto: CreateLinkDto) {
     try {
-      return await this.prisma.link.create({ data: createLinkDto });
+      if (await this.checkUrl(createLinkDto.url)) {
+        return await this.prisma.link.create({ data: createLinkDto })
+      } else {
+        throw new BadRequestException('The url you entered is not found!')
+      }
     } catch (e) {
-      if (e instanceof Prisma.PrismaClientKnownRequestError) {
-        if (e.code === 'P2002') {
-          throw new BadRequestException('This slug violates the unique constraint!');
-        }
+      if (e instanceof Prisma.PrismaClientKnownRequestError && e.code === 'P2002') {
+        throw new BadRequestException('This slug violates the unique constraint!')
       }
+      throw e
     }
   }
 
-  findAll() {
-    return this.prisma.link.findMany();
+  async findAll(params: any) {
+    return this.prisma.link.findMany({
+      where: {
+        title: {
+          contains: params.title,
+        },
+      },
+    })
```

Comment:
This can be improved by making it case-insensitive:
[https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#case-insensitive-filtering](https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#case-insensitive-filtering)